src/gui/widgets/EffectView.cpp

```diff
@@ -90,31 +93,36 @@ EffectView::EffectView( Effect * _model, QWidget * _parent ) :
 		QFont f = ctls_btn->font();
 		ctls_btn->setFont( pointSize<8>( f ) );
 		ctls_btn->setGeometry( 140, 14, 50, 20 );
-		connect( ctls_btn, SIGNAL( clicked() ),
-					this, SLOT( editControls() ) );
+		connect(ctls_btn, SIGNAL(clicked()), this, SLOT(editControls()));
 
 		m_controlView = effect()->controls()->createView();
-		if( m_controlView )
+		if (m_controlView)
 		{
-			m_subWindow = gui->mainWindow()->addWindowedWidget( m_controlView );
-
-			if ( !m_controlView->isResizable() )
+			if (dynamic_cast<PeakControllerEffectControlDialog*>(m_controlView) == nullptr)
```

Comment:
I think it's the trickiest part of this pull request. If we can move some GUI logic to `EffectControlDialog` or some effect-dependent places, the code will be much cleaner than now.

Reply:
Do you mean something like 'isPeakController()'?

Reply:
I'm not talking about that part. I mean the whole logic that shows the dialog.
My idea is making the current behavior default for `EffectControlDialog`(or something related to effect classes) and overriding it in the peak controller effect.

Reply:
Okay, I moved the logic from EffectView to EffectControlDialog.